FHEM/14_SD_WS.pm

```diff
@@ -440,14 +440,12 @@ sub SD_WS_Parse($$)
 		my $checkTemp1 = SD_WS_binaryToNumber($binvalue, 53, 55) ^ 0b111;
 		my $checkTemp2 = SD_WS_binaryToNumber($binvalue, 56, 59) ^ 0b1111;
 		my $checkTemp3 = SD_WS_binaryToNumber($binvalue, 60, 63) ^ 0b1111;
-		$temp = $temp1Dec.$temp2Dec.".".$temp3Dec;
-		
 		if ($checkTemp1 != $temp1Dec || $checkTemp2 != $temp2Dec || $checkTemp3 != $temp3Dec)
 		{
 			Log3 $iohash, 4, "$name: SD_WS_Parse BresserTemeo checksum error in Temperature";
 			$checksumOkay = 0;
 		}
-
+		$temp = sprintf("%.1f", $temp1Dec.$temp2Dec.".".$temp3Dec);
```

Comment:
Du verwendest sprintf mit drei Variablen, aber nur einem Formart? Ich hab gerade keine Ahnung wieso.

Ich vermute es soll so etwas wie 20.4 heauskommen können?  Da ich nicht weiss, wo das Vorzeichen her kommt, nehme ich jetzt mal an, alle drei Variablen sind unsigned integer:

Wieso dann nicht mittels sprintf die drei dezimalwerte zusammenbauen?
$temp=sprintf("%d%d.%d", $temp1Dec,$temp2Dec,$temp3Dec);

Reply:
Ich gehe mal davon aus, das die drei Variablen (unsigned int) zuerst verbunden und dann erst formatiert werden. Das Vorzeichen (negativ) wird erst später ergänzt.

Reply:
Deine Variante macht genau das daraus, was wir nicht wollten:
T: 03.2 H: 68
Meine Variante bringt das gewünschte Ergebis:
T: 3.2 H: 68

Reply:
Okay, jetzt habe ich es verstanden :)

Ein einfaches suchen Ersetzen wäre leichter zu lesen.
`$temp =~ s/^0//;`

Was ich an dem `$temp1Dec.$temp2Dec.".".$temp3Dec` einfach schwer zu lesen finde ist, dass halt hier Zahlen zusammengesetzt werden. Dass das überhaupt funktioniert wundert mich, denn ich wäre davon ausgegangen, dass man hier rechnen muss:
Weisst Du warum das geht, also wieso es wie zahlen gewertet wird?

```
$temp1Dec=temp1Dec*10;
$temp3Dec=$temp3Dec/10;

```

Reply:
Ich habe den Code für das Protokoll 44 nicht geschrieben. Da sich bisher keiner beschwert hat, gehe ich davon aus, das es auch bisher funktioniert hat. Die Zuusammensetzung war bis jetzt so gelöst:
$hum = $hum1Dec.$hum2Dec;
$temp = $temp1Dec.$temp2Dec.".".$temp3Dec;
@Ralf9 hatte in https://github.com/RFD-FHEM/RFFHEM/commit/614a332accbd448ee9355cd2b2d2e65645a715c0#r31370121 auf die führende Null hingewiesen und ich habe es halt korrigiert und getestet.

Reply:
Dann hau doch einfach nur ein $temp =~ s/^0//; mit rein und gut ist :)

Reply:
Wenn dir soviel daran liegt :-)